packages/ruby/lib/readme/har.rb

```diff
@@ -1,11 +1,139 @@
+require "rack"
+require "rack/request"
+
 module Readme
   class Har
-    def initialize(env)
-      @env = env
+    VERSION = "1.2"
+    HTTP_NON_HEADERS = [
+      Rack::HTTP_COOKIE,
+      Rack::HTTP_VERSION,
+      Rack::HTTP_HOST,
+      Rack::HTTP_PORT
+    ]
+
+    def initialize(env, status, headers, response, start_time)
+      @env = Rack::Request.new(env)
+      @response = Rack::Response.new(response, status, headers)
+      @start_time = start_time
     end
 
     def to_json
-      File.read(File.expand_path("../../har.json", __FILE__))
+      {
+        log: {
+          version: VERSION,
+          creator: creator,
+          entries: entries
+        }
+      }.to_json
+    end
+
+    private
+
+    def creator
+      {
+        name: Readme::Metrics::SDK_NAME,
+        version: Readme::Metrics::VERSION
+      }
+    end
+
+    def entries
+      [
+        {
+          cache: {},
+          timings: timings,
+          request: request,
+          response: response,
+          startedDateTime: @start_time.iso8601,
+          time: elapsed_time
+        }
+      ]
+    end
+
+    def timings
+      {
+        send: 0,
+        receive: 0,
+        wait: elapsed_time
+      }
+    end
+
+    def elapsed_time
+      ((Time.now - @start_time) * 1000).to_i
+    end
+
+    def request
+      {
+        method: @env.request_method,
+        queryString: to_hash_array(@env.GET),
+        url: @env.url,
+        httpVersion: @env.get_header("HTTP_VERSION"),
+        headers: http_headers,
+        cookies: cookies,
+        postData: {
+          text: request_body,
+          mimeType: @env.content_type
+        },
+        headersSize: -1,
+        bodySize: @env.content_length&.to_i || -1
+      }
+    end
+
+    def http_headers
```

Comment:
Agreed. I wonder if it might make sense to pull out some sort of decorator over `Rack::Request`. I think I might do something like that in a follow-up PR